### PR TITLE
I18N-1028 - Fixed bug that occurred when same strings were added by multiple PRs

### PR DESCRIPTION
**Pull request by DarKhaos**
The Repository Statistics are generated even when the branch does not have new Text Units

### CODE REVIEW

**Comment by maallen** [comment]
Please add a description of the changes

File: webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java [hunk 1]
```diff
@@ -165,9 +167,17 @@ public class AssetExtractionService {
 
   @Autowired LocalBranchToEntityBranchConverter localBranchToEntityBranchConverter;
 
+  private RepositoryStatisticsJobScheduler repositoryStatisticsJobScheduler;
```
Comment (maallen) [inline webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java:170]: You should be able to use `@Autowired` here directly?

File: webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java [hunk 2]
```diff
@@ -165,9 +167,17 @@ public class AssetExtractionService {
...
+
   @Value("${l10n.assetExtraction.quartz.schedulerName:" + DEFAULT_SCHEDULER_NAME + "}")
   String quartzSchedulerName;
 
+  @Autowired
```
Comment (maallen) [inline webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java:175]: I don't believe this setter here is necessary

File: webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java [hunk 3]
```diff
@@ -444,6 +454,11 @@ CreateTextUnitsResult createTextUnitsForNewContent(
           ImmutableList<BranchStateTextUnit> toCreateTmTextUnits =
               getBranchStateTextUnitsWithoutId(stateForNewContentWithIds);
 
+          final Repository repository = assetContent.getBranch().getRepository();
+          if (toCreateTmTextUnits.isEmpty() && repository != null) {
```
Comment (maallen) [inline webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java:458]: Is the intention here that if `toCreateTmTextUnits` is empty that means that the branch will not show in the branch page as no text units are created so no stats job is triggered?

Can you put a comment in the code explaining this if that's the case

File: webapp/src/test/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionServiceTest.java [hunk 4]
```diff
@@ -1365,6 +1370,51 @@ public void testTextUnitsWithMultipleUsagesForPlural() throws Exception {
     checkAssetTextUnits(assetTextUnits, expectedUsages);
   }
 
+  @Test
+  public void testBranchesWithSameStrings() throws Exception {
+    final RepositoryStatisticsJobScheduler mockedScheduler =
+        Mockito.mock(RepositoryStatisticsJobScheduler.class);
+    this.assetExtractionService.setRepositoryStatisticsJobScheduler(mockedScheduler);
```
Comment (maallen) [inline webapp/src/test/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionServiceTest.java:1377]: Ah I think I see why you'd the set method earlier, if you Autowire the bean and don't set it as a private field it'll be protected access, so you can set it directly in the test as it's the same package using `this.assetExtractionService.repositoryStatisticsJobScheduler = mockedScheduler`

Comment (DarKhaos) [reply webapp/src/test/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionServiceTest.java:1377]: I tried that, but the mock object was not being used by the AssetExtractionService object, and this assert `verify(mockedScheduler, times(1)).schedule(anyLong());` always failed

Comment (DarKhaos) [reply webapp/src/test/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionServiceTest.java:1377]: **@MockBean** and **@SpyBean** caused several errors with the following message:
`org.springframework.jdbc.CannotGetJdbcConnectionException: Failed to obtain JDBC Connection`

Comment (DarKhaos) [reply webapp/src/test/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionServiceTest.java:1377]: > I would still expect that you're able to Autowire the RepositoryStatisticsJobScheduler in the AssetExtractionService via field injection, drop the private access modifier against it and then you should be able to set the field directly in your test (not using a @MockBean, just using a standard mock as you do in the set method approach).
After making that change, I am getting this error message:

> ERROR] Failures: 
[ERROR] com.box.l10n.mojito.service.assetExtraction.AssetExtractionServiceTest.testBranchesWithSameStrings
[ERROR]   Run 1: AssetExtractionServiceTest.testBranchesWithSameStrings:1419 
Wanted but not invoked:
repositoryStatisticsJobScheduler.schedule(
    <any long>
);
-> at com.box.l10n.mojito.service.repository.statistics.RepositoryStatisticsJobScheduler.schedule(RepositoryStatisticsJobScheduler.java:21)
Actually, there were zero interactions with this mock.

[ERROR]   Run 2: AssetExtractionServiceTest.testBranchesWithSameStrings:1419 
Wanted but not invoked:
repositoryStatisticsJobScheduler.schedule(
    <any long>
);
-> at com.box.l10n.mojito.service.repository.statistics.RepositoryStatisticsJobScheduler.schedule(RepositoryStatisticsJobScheduler.java:21)
Actually, there were zero interactions with this mock.
For some reason, setting the field directly in the test did not work; the mock object is not being used during execution, but the real object is.

File: webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java (hunk 1 above)
Comment (DarKhaos) [reply webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java:170]: Please, read comment https://github.com/pinterest/mojito/pull/116#discussion_r1575394017

File: webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java (hunk 2 above)
Comment (DarKhaos) [reply webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java:175]: Please, read comment https://github.com/pinterest/mojito/pull/116#discussion_r1575394017

**Review by maallen — APPROVED** [review]

**Pull request merged**